design/specs/list-data-type.md

```diff
@@ -0,0 +1,194 @@
+
+### 1.1 Viewing Existing List Data
+
+Existing Mathesar List data distinguishes itself from other data type through four unique ways: 
+
+1. Column header data type icon.
+2. Data type icon and description in the column header menu. `Allow List` option is required to be turned on for List data type.
+3. Tooltip information when hovering on the column header data type icon. 
+4. Items in a single cell is separated by comma delimiters and a `show all` option is present. Users can use `show all` function to expand the List and view all List items.
+
+---
+Wireframes
+
+[Viewing Existing List Data](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=7317%3A80859&viewport=306%2C48%2C0.59&scaling=min-zoom&starting-point-node-id=7379%3A82626&show-proto-sidebar=1)
+
+---
+
+### 1.2 Filtering Existing List Data
+
+The following filters are available for List data:
+
+1. is empty
+2. is not empty
+3. contains `<LIST ITEM DATA TYPE>`
+4. number of items greater than `<NUMBER>`
+5. number of items greater than or equal to `<NUMBER>`
+6. number of items equal to `<NUMBER>`
+7. number of items lesser than `<NUMBER>`
+8. number of items lesser than or equal to `<NUMBER>`
+
+Users are able to utilize multiple filters through the `Add Filter` feature. By default, the relationship between filters is set to `difference`. Users can enable the `Union` toggle to include data cells that appear in any of the filters.
+
+---
+Wireframes
+
+[Filtering Existing List Data](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=7317%3A80859&viewport=306%2C48%2C0.19&scaling=min-zoom&starting-point-node-id=7379%3A82944&show-proto-sidebar=1)
+
+---
+
+### 1.3 Editing Existing List Data
+
+Users can single-click a List data cell or click the `show all` button to expand the selected the data cell and enter editing mode. A background shade is added to each List item, not the delimiters, to highlight the nature of List data type. Under cell editing mode, users can again single-click the item that they wish to edit to finish the edit without compromising the data structure.
+
+To add a new List item in the cell, users can simply single-click the white space within the cell when under editing mode. A new item placeholder and a new delimiter will show up. A preset Null value will be entered automatically if users have allowed and configured it.
+
+---
+Wireframes
+
+[Editing Existing List Data](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=7317%3A80859&viewport=306%2C48%2C0.24&scaling=min-zoom&starting-point-node-id=7809%3A84002&show-proto-sidebar=1)
+
+---
+
+### 1.4 Editing Existing List Data – Item Type Warning
+
+While editing List data items, if Mathesar detected that users were entering a potentially non-conforming item type (e.g., entering a number into a List of Text items), a warning modal would show up to inform users. 
```

Comment:
It would be better to have a different example here, numbers are text too. For example, a list of movies would have a list item type of Text but could include `9`, [which is a movie](https://www.imdb.com/title/tt0472033/).

Adding text into a list of numbers would be a better example. We definitely know when something is not a number.

Reply:
Agreed. The feature explored here is to allow Mathesar to inform users if it thinks users could have made a mistake, and it could happen to all basic data types. Mathesar would know whether 9 is a text or a number, but it could ask users to double-check when non-conforming items are detected.